src/main/java/org/broadinstitute/hellbender/tools/spark/utils/LongBloomFilter.java

```diff
@@ -0,0 +1,245 @@
+        for (int i = 0; i < legalSizes.length; i++) {
+            if (minNumBytes <= legalSizes[i]) {
+                return legalSizes[i];
+            }
+        }
+        throw new IllegalArgumentException("No legal sizes large enough for size " + minNumBytes);
+    }
+
+    public static int getLegalSizeBelow(final long maxNumBytes) {
+        if (maxNumBytes <= legalSizes[0]) {
+            throw new GATKException("No legal sizes smaller than size " + maxNumBytes);
+        }
+        for (int i = 1; i < legalSizes.length; i++) {
+            if (maxNumBytes < legalSizes[i]) {
+                return legalSizes[i - 1];
+            }
+        }
+        return legalSizes[legalSizes.length - 1];
+    }
+
+    public static long getOptimalNumberOfBytes(final long num_elements, final double fpp) {
+        return (long) Math.ceil(-num_elements * Math.log(fpp) / (8 * Math.log(2) * Math.log(2)));
+    }
+
+    private long countBits() {
+        long X = 0;
+        for (final byte b : buckets) {
+            byte bb = b;
+            for (int i = 0; i < Byte.SIZE; i++) {
+                X += bb & 1;
+                bb >>= 1;
+            }
+        }
+        return X;
+    }
+
+    private long estimateSize(final long numBitsSet) {
+        return (long) (-(((double) numBits) / numHashes) * Math.log(1 - (((double) numBitsSet) / numBits)));
+    }
+
+    protected void serialize(final Kryo kryo, final Output output) {
+        final boolean oldReferences = kryo.getReferences();
+        kryo.setReferences(false);
+
+        output.writeLong(numBits);
+        output.writeInt(numBuckets);
+        output.writeInt(numHashes);
+        output.writeBytes(buckets);
+
+        kryo.setReferences(oldReferences);
+    }
+
+    public boolean add(final long entryValue) {
+        final long[] bitIndex = entryIndices(entryValue);
+        for (int i = 0; i < bitIndex.length; i++) {
+            final int bucket = indexBucket(bitIndex[i]);
+            buckets[bucket] |= bucketMask(bitIndex[i]);
+        }
+        return true;
+    }
+
+    public void addAll(final long[] entryValues) {
+        for (final long val : entryValues) {
+            add(val);
+        }
+    }
+
+    /**
+     * Using different semantics than a Set because false positives are possible
+     */
+    public boolean contains(final long key) {
+        final long[] bitIndex = entryIndices(key);
+        for (int i = 0; i < bitIndex.length; i++) {
+            final int bucket = indexBucket(bitIndex[i]);
+            if ((bucketMask(bitIndex[i]) & buckets[bucket]) == 0) return false;
+        }
+        return true;
+    }
+
+    public boolean containsAll(final long[] vals) {
+        for (final long val : vals) {
+            if (!contains(val))
+                return false;
+        }
+        return true;
+    }
+
+    /**
+     * Determines partition corresponding to the given the bit index
+     */
+    private int indexBucket(final long bitIndex) {
```

Comment:
I've done away with this method but its replacements bitIndexToBucketArray() and bitIndexToBucketIndex() cannot be static